crates/neon/src/types_impl/mod.rs

```diff
@@ -1173,7 +1170,7 @@ impl<CL: Object> JsFunction<CL> {
     /// Calls this function as a constructor.
     ///
     /// **See also:** [`JsFunction::construct_with`].
-    pub fn construct<'a, 'b, C: Context<'a>, AS>(&self, cx: &mut C, args: AS) -> JsResult<'a, CL>
+    pub fn construct<'a, 'b, C: Context<'a>, AS>(&self, cx: &mut C, args: AS) -> JsResult<'a, JsObject>
```

Comment:
Goodbye soundness hole!

_Technically it wasn't a soundness hole because Node-API **also** checks the type, but at least it won't unexpectedly crash._